### ISSUE TITLE
Is this project retired?

### ISSUE THREAD

**Issue by mikestopcontinues** [keystonejs/keystone#4434]
I really like Keystone, but I don't want to commit my next site to a retired library. What's the plan going forward? Are you looking for new maintainers? Thanks!

**Comment by christroutner** [comment]
I've built dozens of sites with KeystoneJS, while using very few of the updates made in v4.0. Why does this project need to keep growing to be useful? It's the code in the project that is the most stable, that provides the most value to me, as a website and web app developer.

**Comment by mikestopcontinues** [comment]
Because they are halfway to a version upgrade with a lot of cool new features and breaking changes. There are a lot of open issues and pull requests, and no meaningful updates in a while. I wouldn't be very responsible to my future self if I invested in a project without leadership when I could choose another or roll my own. I'm not looking to do either, but if I have to, I have to...

**Comment by Takumiiino** [comment]
@christroutner 

So, how do you implement "User role" or "Permission" in AdminUI for your dozens of sites? E.g. only admin can edit specific model, default filter for non-admin user to protect or hide others item. I googled about a month and i still stuck about permission problem. Do you have any hints? 

Already checked those issue or pull request:

#803  
https://github.com/keystonejs/keystone-test-project/pull/21
https://github.com/keystonejs/keystone/issues/334

etc...
none of them save me.

**Comment by Takumiiino** [comment]
@christroutner 

Thanks for your reply. I have already read some info of ConnextCMS.
However, "multiple CMS" in my opinion is not a good solution to deal with Permission/User Role. Adding frameworks or extensions sometimes are worth because of saving time, code less etc... but if the frameworks are too large or complicated, they are not worth spend much time to implement within a project which already in a complicated framework, like keystone.

All we need is a permission system which base on keystone model, keep it clear and simple like hidden: true, defaultSort: 'name'. 

Maybe something like:
```
var aList = new keystone.List('aList', {
	track: true,
	defaultFilter: {'createdBy': _req_user}
});
```
Or
what @JedWatson mentioned in https://github.com/keystonejs/keystone-test-project/pull/21

Please, if any Contributor still here, please complete those issues.

**Comment by christroutner** [comment]
@Takumiiino why can't you be the contributor? Create a change, submit a pull request. That's how changes get made to open source projects.

**Comment by mikestopcontinues** [comment]
@christroutner There's only been a trickle of accepted PRs in the last six months, and plenty to choose from. Most have been ignored. Most accepted have been trivial. That doesn't inspire participation.

**Comment by JedWatson** [comment]
Sorry for the lack of communication. Despite the recent trickle of public contributions, we're absolutely committed to the project and it's not going away.

(There are, unsurprisingly, reasons for the lack of progress in the last six months but I'm not going to save that for a post I'm writing on medium at the moment rather than making this a really long reply)

We are still progressing towards a final release of 4.0 and have been putting together a roadmap I'll link separately so everyone can follow along. We've also learned a lot in the process and have been working on prototypes for the future that we wanted to flesh out more completely before sharing (I really don't like promising things for the future, prefer showing things that are real) but obviously the lack of commuication hasn't been a great alternative.

After 4.0 we have some big plans for Keystone that are already in progress and am really excited to start sharing information (including some new contributors who will be coming on board to help manage and drive the project) but we need flesh it out a bit more first.

One of the highest priorities for the project has always been backwards compatibility and/or a smooth upgrade path between versions - sometimes, honestly, to our detriment because this has frequently taken dev effort that probably could have been better invested moving forward - but it's still a massive commitment. Given that almost *everything* was rewritten between v3 and v4, and there were really only minor backwards incompatible changes (e.g. the new File field) I think our history backs that up and is part of why people are confident building on Keystone.

It's probably also worth mentioning we would love to revitalise the community and invite more contributors once we can be organised about how to best manage it and not waste anybody's time.

So in short, no, Keystone isn't dead or abandoned, we’ve just been really busy. We still use Keystone extensively ourselves and have no plans to let it languish.

**Comment by Kalcode** [comment]
Appreciate your follow-up. I've enjoyed working with this CMS and feel its only lacking a few things here and there that would really help it stand out. Really glad it not being forgotten about.

**Comment by mikestopcontinues** [comment]
Thanks, Jed! That's above and beyond what I was hoping. I'll be watching closely for any way I can pitch in. Also, I consider this thread closed, but I'll leave it open just in case you'd like to get a few more eyeballs on your response. Thanks again!

**Comment by VinayaSathyanarayana** [comment]
@JedWatson - Appreciate it. We all love Keystone.

**Comment by areknow** [comment]
@JedWatson Thats great news, thanks for the update! We are all looking forward to the release and the future of Keystone.

**Comment by teamhail** [comment]
Jed, awesome to hear from you.  Hope all is well with everyone.  Terrific project and thanks very much for everyone's effort.

**Comment by w0rldart** [comment]
@JedWatson thanks for the update. I was wondering how are you planning to tackle the currently open 116 pull requests and 296 issues now, and for future term? Will the new contributors take care of that, or is something that you are looking to keep internal?

**Comment by Elemino** [comment]
I have been learning keystone for almost a year now and am very looking forward to seeing this project grow and become a legendary piece of software.  I think there's nothing like keystone on the web. Keep up the great work.

**Comment by Noviny** [comment]
I'm going to leave this issue open until we have Jed's post.

Just wanted to say, reading everyone's support and good wishes has been a real boost. I'm working to spin up some processes to get 4 out, which I'll get a separate post up about, for those wanting to help, or wanting visibility.

**Comment by VinayaSathyanarayana** [comment]
Waiting for @JedWatson 's post on the go forward plans.

**Comment by JedWatson** [comment]
Another update - we've been working to organise the backlog of work required to get v4 out the door, and are getting close to being able to communicate our status and roadmap more clearly.

To answer @w0rldart's question directly, it's a lot of work and we're not intentionally keeping it internal, so if anybody can dedicate some time to helping us please let me know and we'll figure out how to co-ordinate our efforts.

I'm going to close this now, because I think we've answered the original question. I'm thrilled so many people care about the project and look forward to involving the community better in the future 🙂

**Comment by VinayaSathyanarayana** [comment]
Was the goforward plan published?

**Comment by simonwidjaja** [comment]
Is the medium article already out? Did I miss it? Or any other news regarding the future of Keystone?

**Comment by afilp** [comment]
@JedWatson Can you please give us a hint if the user roles (very important!) is to be implemented very soon for 4.0? I hold on using Keystone for a project, just for that. I will have to use a different solution (don't like) if I know that this may take 3 months+. Thanks!

**Comment by VinayaSathyanarayana** [comment]
@JedWatson - Requesting an update on the go forward plans

**Comment by MikeLimeRocket** [comment]
@afilp Do you require roles for publishing workflow i.e. author, editor etc?  There is a sponsored publishing-workflow feature branch here https://github.com/keystonejs/keystone/tree/publishing-workflow that was worked on recently that might have enough there for your immediate needs. It includes basic roles, publishing workflow and revisions.

**Comment by simonwidjaja** [comment]
@JedWatson - Requesting an update on the go forward plans

**Comment by VinayaSathyanarayana** [comment]
@JedWatson - Please publish go forward plan

**Comment by gall0** [comment]
@MikeLimeRocket Are the revisions displayed in the Admin UI? Are you able to hide lists based on roles? Thanks in advance!

**Comment by MikeLimeRocket** [comment]
@gall0 yes revisions and the ability to restore are displayed in the Admin UI. For hiding entire lists I don't think that was included in this branch. You can restrict users to not being to edit or publish though if that satisfies your needs?

**Comment by gall0** [comment]
Thank you so much for the prompt response!

----- Original Message -----

**Comment by mikestopcontinues** [comment]
In the JS ecosystem, Keystone is still our best hope for rapidly developing websites as easily as with Wordpress, but without all the PHP and Wordpress BS. But at the same time, the lack of development on the project keeps me from actually wanting to use it in production, whether for my own sites or clients. I would imagine it's the same for lots of people watching this project. And I would guess that lots of people here would also be interested in contributing—if there was a concrete direction and concrete leadership. But 74 outstanding PRs reads to everyone like it would be wasted work to even bother.

Anyway, I'm still watching Keystone, and I'd really like to see JS finally get that game-changing CMS.

**Comment by internetErik** [comment]
I worth mentioning that Keystone 5 is in active development. The development is very active and ongoing in a private repository.

**Comment by ttsirkia** [comment]
Is there a specific reason why the development is in a private repo instead of a public branch? In this way, no feedback/help/discussion from the community.

**Comment by internetErik** [comment]
A good place to ask about this (and for access to the repository, too) would be on the community slack.

**Comment by ttsirkia** [comment]
So you mean that the repository is not company private but there are some community members as well?

**Comment by internetErik** [comment]
Yeah, they seem to let people have access to the repository without too much fuss if you ask. But the deal is they trust you won't share it in a public repo
